File: lakshya_46/frontend/lib/models/book.dart
class Book {
  String id;
  String title;
  String author;
  String genre;
  String price;
  String publishedYear;
  String? createdDate;

  Book({
    required this.id,
    required this.title,
    required this.author,
    required this.genre,
    required this.price,
    required this.publishedYear,
    this.createdDate,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      author: json['authorName'] ?? '',
      genre: json['genre'] ?? '',
      price: json['price'] ?? '',
      publishedYear: json['publishedYear'] ?? '',
      createdDate: json['createdDate'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'authorName': author,
      'genre': genre,
      'price': price,
      'publishedYear': publishedYear,
    };
  }
}

File: lakshya_46/backend/models/Book.js
const mongoose = require('mongoose');

const bookSchema = new mongoose.Schema({
    title: {
        type: String,
        required: true,
    },
    authorName: {
        type: String,
        required: true,
    },
    genre: {
        type: String,
        required: true,
    },
    price: {
        type: String,
        required: true,
    },
    publishedYear: {
        type: String,
        required: true,
    },
    createdDate: {
        type: Date,
        default: Date.now,
    }
});

module.exports = mongoose.model('Book', bookSchema);